nestauk/dsp_interview_transcripts
# Add data processing steps, topic modelling, tests, streamlit app

This PR provides some options for how we could preprocess the data, generates clusters with BERTopic from each of the datasets, and provides a streamlit app to visualise the clusters.

The script `dsp_interview_transcripts/pipeline/process_data.py` creates two datasets: one contains user messages more than 4 words long, and one contains each user message and the bot message immediately preceding.

The script `python dsp_interview_transcripts/pipeline/segment_interviews.py` compares the questions from the topic guide and the actual messages from the bot. Once a match is found, this is forward-filled over the subsequent user and bot messages up until the next bot message that matches one of the questions in the topic guide.

The only hyperparameter we've varied so far for the topic modelling is the minimum cluster size.

Still to do: generating more helpful cluster summaries!

**To test this**
- Run `python dsp_interview_transcripts/pipeline/run_pipeline.py`
- Run `streamlit run dsp_interview_transcripts/pipeline/app.py`
- Run `pytest` (should run all the tests in the `tests/` directory)

**Please pay attention to**
- Do the different ways of cutting up the text make sense and how could we improve on these? (Though this might be hard to answer without seeing helpful topic names and summaries)
